R/preprocessing_pipeline.R

```diff
@@ -0,0 +1,34 @@
+preprocessing_pipeline <- function(choices  ,
+                                   text,
+                                   infrequent_term_threshold = .01,
+                                   verbose = FALSE,
+                                   save_dfm = FALSE,
+                                   intermediate_directory = NULL)
+{
+    stopwords <- if(choices$removeStopwords) quanteda::stopwords() else FALSE
+    if(choices$use_ngrams){ #need to tokenize before ngram before preprocessing
```

Comment:
Putting it in this order will form ngrams of un-stemmed words, but then later if `stem = TRUE` then it will stem the last part (only) of the ngram. Might make sense first to stem using `tokens_wordstem()`, which also - unlike the dfm option - provides a language choice.

Reply:
Thanks, I completely forgot to account for the order.